File: food_app/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final String image;
  final double price;
  final double rating;
  final String calories;
  final String category;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.image,
    required this.price,
    required this.rating,
    required this.calories,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'].toString(), // ✅ MongoDB ObjectId
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      image: json['image'] ?? '',
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
      rating: (json['rating'] as num?)?.toDouble() ?? 4.0,
      calories: json['calories'] ?? '',
      category: json['category'] ?? 'Classics',
    );
  }
}

File: foodapp-backend/src/models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },

    description: {
      type: String,
      default: "",
    },

    // ✅ STORE IMAGE URL (placeholder allowed)
    image: {
      type: String,
      default: "https://via.placeholder.com/300",
    },

    price: {
      type: Number,
      required: true,
    },

    rating: {
      type: Number,
      default: 4.0,
    },

    calories: {
      type: String,
      default: "",
    },

    // ✅ MATCH UI CATEGORIES
    category: {
      type: String,
      enum: ["Combos", "Sliders", "Classics"],
      default: "Classics",
    },

    isAvailable: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Product", productSchema);
